### PR TITLE
Adds Staff_Code to In-House Repair.

### PR DESCRIPTION
**Pull request by scherztc**
Fixes : #276

### CODE REVIEW

File: spec/features/admin_user_spec.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
 require 'rails_helper'
 
 RSpec.describe 'Admin User Tests', type: :feature do
+  let(:staff_code) { create(:staff_code, code: 'C', points: 10) }
```
Comment (scherztc) [inline spec/features/admin_user_spec.rb:6]: @hortongn : Can someone help me get this test working?

**Comment by crowesn** [comment]
Looks like this has a conflict. Go ahead and rebase, I will start reviewing.

**Comment by crowesn** [comment]
@scherztc Is staff_code a required field for in_house_repairs?

**Comment by scherztc** [comment]
> @scherztc Is staff_code a required field for in_house_repairs?

Yes.  It acts that way.  There is also an asset generator that you should run in the beginning.  

rails batch:staff_code_controlled_vocabulary

**Review by crowesn — CHANGES_REQUESTED** [review]
1. Use cancan to restrict StaffCode crud to admin role (maybe standard?) in  `app/models/ability.rb`
2. Require authentication in the StaffCode controller
    > `before_action :authenticate_user!`
    > `load_and_authorize_resource`
5. Once we've decided who should be permitted to crud StaffCode records; if admin only, make sure to edit the standard_user feature spec to test that they *cannot* crud.
4. Add test to read_only user feature spec to confirm that read_only users cannot crud StaffCodes
6. Remove `app/assets/stylesheets/scaffolds.scss`
7. Add validation check for in_house_repair_record in `app/controllers/in_house_repair_records_controller.rb` (see inline comment)

**Comment by crowesn** [comment]
@scherztc 

Coveralls report shows several lines in the staff codes controller aren't being touched by any specs. Also wants a test for a fails validation on on the :in_house_repair. 

https://coveralls.io/builds/49070534/source?filename=app%2Fcontrollers%2Fstaff_codes_controller.rb#L33
https://coveralls.io/builds/49070534/source?filename=app%2Fcontrollers%2Fin_house_repair_records_controller.rb#L14

**Review by crowesn — APPROVED** [review]
Nice work.

**Pull request merged**